lib/logproto/logproto-framed-server.c

```diff
@@ -213,9 +225,103 @@ log_proto_framed_server_fetch(LogProtoServer *s, const guchar **msg, gsize *msg_
               self->buffer_end = self->buffer_end - self->buffer_pos;
               self->buffer_pos = 0;
             }
-          goto read_message;
+          if (self->state == LPFSS_TRIM_MESSAGE)
+            goto trim_message;
+          else
+            goto read_message;
+        }
+      break;
+
+    case LPFSS_TRIM_MESSAGE:
+
+trim_message:
+      if (self->buffer_end == self->buffer_size)
```

Comment:
Man, you have nailed it. This is a dead code. Because without reading first, the buffer_end will never be equal to the buffer_size. (Actually we will not create a chunked message, because there is a memmove at the end of frame_read, but this condition will NEVER be true.)